### PR TITLE
Fix navigation

### PR DESCRIPTION
**Pull request by DanielHeath**

### CODE REVIEW

File: src/pages/PersonalisationWizardPage.js [hunk 1]
```diff
@@ -21,6 +21,15 @@ class PersonalisationWizardPage extends BasePersonalisationPage {
         var subpage = this.state.subpage - 1;
 
         if (subpage < 0) {
+            /*
+             * FIXME: If we use a real URL for each subpage,
```
Comment (DanielHeath) [inline src/pages/PersonalisationWizardPage.js]: @danni I'd love to hear your thoughts on this approach to getting the back/forwards to work nicely in personalisation.

I think it'll let us use regular history while still preserving the desired behavior of having the `back` button skip over the personalisation questions.

**Comment by DanielHeath** [comment]
++?

File: src/client.js [hunk 2]
```diff
@@ -10,7 +10,20 @@ import routes from "./routes";
 // For onTouchTap: see https://github.com/zilverline/react-tap-event-plugin
 injectTapEventPlugin();
 
+/*
+ * If at any point there isn't a meaningful 'back',
+ * go to the app homepage instead.
+ */
 let history = createBrowserHistory();
+const goBack = history.goBack;
```
Comment (danni) [inline src/client.js]: Can we do this properly by inheriting the history class from ReactRouter ?

File: src/pages/BaseResultsPage.js [hunk 3]
```diff
@@ -107,9 +107,15 @@ class BaseResultsPage extends BaseCategoriesPage {
...
+                const sep = this
+                    .props
+                    .location
+                    .pathname
+                    .endsWith("/") ? "" : "/";
+
                 this.history.replaceState(
                     null,
-                    this.props.location.pathname + "/personalise",
+                    this.props.location.pathname + sep + "personalise",
```
Comment (danni) [inline src/pages/BaseResultsPage.js]: Template string?

File: test/steps/browser.js [hunk 4]
```diff
@@ -26,6 +26,15 @@ module.exports = (function() {
               unpromisify(doSearchAndEnter))
         .when("I click on the search icon", unpromisify(clickSearchIcon))
         .when("I click back from the title bar", unpromisify(clickBack))
+        .when(
+            "I click back from the browser UI",
+            unpromisify(clickBrowserBack),
+        )
+        .when(
+            "I click back from the top bar",
+            unpromisify(clickAppBarBack),
```
Comment (danni) [inline test/steps/browser.js]: This already exists I think?

File: src/pages/PersonalisationSummaryPage.js [hunk 5]
```diff
@@ -10,23 +10,15 @@ import icons from "../icons";
...
-            this.props.history.goBack();
-        } else {
-            this.setState({subpage: 0});
-        }
+        this.props.history.goBack();
     }
 
     nextStep(): void {
-        this.setState({subpage: 0});
+        this.props.history.goBack();
```
Comment (danni) [inline src/pages/PersonalisationSummaryPage.js:17]: Can't we just call goBack from the `NavigationMixin` ?

File: test/steps/browser.js (hunk 4 above)
Comment (DanielHeath) [inline test/steps/browser.js]: Indeed it does, thanks.

File: src/pages/PersonalisationSummaryPage.js (hunk 5 above)
Comment (DanielHeath) [inline src/pages/PersonalisationSummaryPage.js:17]: `NavigationMixin` was replaced with `History` mixin in the new react-router, which is what I've used here.

File: src/client.js (hunk 2 above)
Comment (DanielHeath) [inline src/client.js]: It's not a class, but I found a cleaner way to do it nonetheless.

File: src/client.js [hunk 6]
```diff
@@ -10,9 +10,55 @@ import routes from "./routes";
 // For onTouchTap: see https://github.com/zilverline/react-tap-event-plugin
 injectTapEventPlugin();
 
-let history = createBrowserHistory();
+/*
+ * If at any point there isn't a meaningful 'back',
+ * go to the app homepage instead.
+ */
+
+function History() {
```
Comment (danni) [inline src/client.js:22]: Why not use a real class?

**Pull request merged**